recipes/ncbi-cxx-toolkit-public/26/conanfile.py

```diff
@@ -62,6 +62,8 @@ def _translate_ReqKey(self, key):
                 return None
             if key == "NGHTTP2" and self.settings.os == "Windows":
                 return None
+            if key == "MySQL" and self.settings.os == "Macos" and "arm" in self.settings.arch:
```

Comment:
`libmysqlcient` can be built on macOS M1. The issue is cross-build.

Reply:
previous commit failed to build on Macos/arm. for whatever reason.

Reply:
see https://github.com/conan-io/conan-center-index/pull/8802#issuecomment-1049339490

Reply:
so you can replace this condition by the same one in libmysqlclient: https://github.com/conan-io/conan-center-index/blob/96637ea0caca8b7abb61a4142d4c517b4484423d/recipes/libmysqlclient/all/conanfile.py#L120

```suggestion
            if key == "MySQL" and hasattr(self, "settings_build") and tools.cross_building(self, skip_x64_x86=True):
```

Reply:
It turns out that cross compilation is more problematic than I anticipated.
Macos/arm build fails because "datatool" is not found. Datatool is a program - one of the build targets. It generates source code for some projects. Datatool is built before it is needed (of course), but the build system cannot find it.
GRPC solves similar problem by putting PROTOC into a separate package and defining build_requirements.
I wonder if there is a way of telling Conan that build requires a program which is being built by the build itself?
I do not see anything like this in documentation

Reply:
you can put `ncbi-cxx-toolkit-public` into build requirements (see `icu` recipe), but at some point your build system must have a switch to look for an external `Datatool` executable.

Reply:
That is a package build can require itself. Nice.
and the "external datatool" should be in the build's bin directory, probably.
I do not want to make experiments directly in ConanCenter
Probably, the best option at this time is disabling cross-compilation.